themes/default/content/blog/local-testing-with-pulumi/index.md

```diff
@@ -0,0 +1,151 @@
+
+```typescript
+// Build the assets image or pull remote image, depending on value of assetsFlag
+var assetsImageRef: pulumi.Input<string>;
+if (assetsFlag == "build") {
+    const assetsImage = new docker.Image("assets-image", {
+        build: {
+            context: `${srcRepoPath}/src/assets`,
+            dockerfile: `${srcRepoPath}/src/assets/Dockerfile`,
+            platform: imagePlatform,
+        },
+        imageName: "zephyr-assets:latest",
+        skipPush: true,
+    }, { retainOnDelete: true });
+    assetsImageRef = assetsImage.id;
+} else {
+    const assetsRegistryImage = docker.getRegistryImage({
+        name: "public.ecr.aws/aws-containers/retail-store-sample-assets:0.2.0",
+    });
+    const assetsRemoteImage = new docker.RemoteImage("assets-image", {
+        name: assetsRegistryImage.then(assetsRegistryImage => assetsRegistryImage.name),
+        pullTriggers: [assetsRegistryImage.then(assetsRegistryImage => assetsRegistryImage.sha256Digest)],
+    }, { retainOnDelete: true });
+    assetsImageRef = assetsRemoteImage.repoDigest;
+};
+```
+
+{{% notes %}}
+You can see that the released container image is hard-coded in this example. You could also pass this in as a configuration value to allow the Pulumi program to more easily account for new releases.
+{{% /notes %}}
+
+"By setting the flag for the assets service to `build` and all the other services to something else---the code only builds if the value is set to `build`---then I can isolate changes made to that service for testing, to be sure I haven't broken something elsewhere."
+
+"That seems useful."
+
+"Indeed! And you can see here that we are building a Docker network for these containers; this is what enables connectivity between the various parts of the online store when it's running locally via Docker."
+
+```typescript
+// Create a network
+const network = new docker.Network("network", {
+    name: "zephyr-net",
+});
+
+// Create an assets container
+const assetsContainer = new docker.Container("assets-container", {
+    capabilities: {
+        drops: ["ALL"],
+    },
+    envs: [
+        "PORT=8080",
+    ],
+    hostname: "assets",
+    image: assetsImageRef,
+    memory: 64,
+    name: "assets",
+    networksAdvanced: [
+        {
+            name: network.name,
+        },
+    ],
+    restart: "always",
+});
+```
+
+Bob nodded. "Yep, this all makes sense. Interesting, I hadn't really considered Pulumi for this particular use case. Usually when folks mention Pulumi the first thought in my head is provisioning cloud infrastructure."
+
+"Oh, certainly, and it works great for that---as evidenced by the fact that your team, the platform team, uses it to provision all the cloud infrastructure for Zephyr's online store, including [multiple stacks](/blog/iac-recommended-practices-structuring-pulumi-projects/) and [stack references](/blog/iac-recommended-practices-using-stack-references/) to share information across stacks."
```

Comment:
This paragraph feels like it's "laying it on a bit thick" to me. 😄 I think all it really needs to say is "Yep, and it does that well -- but it also works really well for this case, too."

Reply:
Fair comment, thanks! I've reworded it in db276cf.